app/src/main/java/Model/Apartment.java

```diff
@@ -94,26 +101,54 @@ public void setName(String newName) {
      *
      * @param person
      */
-    public void addPersonToApartment(Person person) {
+    public boolean addPersonToApartment( Person person) {
         if (person == null) {
-            return;
+            return false;
+        }
+
+        if (members.contains(person)) {
+            return false;
         }
 
         ParseRelation<Person> relation = getUserRelation();
         relation.add(person);
         incrementNumberOfResidents();
         saveInBackground();
+        return true;
     }
 
-    public void removePersonFromApartment(Person person) {
+    public boolean removePersonFromApartment( Person person) {
         if (person == null) {
-            return;
+            return false;
         }
 
+        if(!members.contains(person)) {
+            return false;
+        }
         ParseRelation<Person> relation = getUserRelation();
         relation.remove(person);
         decrementNumberOfResidents();
         saveInBackground();
+        return true;
+    }
+
+    public void fetchMembersOfApartment(final FindCallback<Person> callback) {
+        final ParseRelation<Person> memberRelation = getUserRelation();
+        ParseQuery<Person> memberQuery = memberRelation.getQuery();
+
+        memberQuery.findInBackground(new FindCallback<Model.Person>() {
+            @Override
+            public void done(List<Person> objects, ParseException e) {
+                if (objects != null) {
+                    members = new ArrayList<>(objects.size());
```

Comment:
Also why does `fetchMembersOfApartment` take an argument of type `FindCallback<Person>` but not use it anywhere.

Also, there's already [a method](https://github.com/ucsdCSE110wi16/CSE110M260T15/blob/InventoryItem/app/src/main/java/Model/Apartment.java#L154,L160) that accomplishes this in the code.